### DISCUSSION TITLE
Renovatebot user does not discover all repositories, although being admin

### DISCUSSION THREAD

**Discussion by manuschillerdev** [renovatebot/renovate#31439] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I use renovatebot with gitlab. it uses a token from a technical user (who is admin) that has the admin_mode permissions. It would suspect that this token should be sufficient to discover all repositories on my private instance. However, the technical user needs to be a member (direct or indirect) member of a project to be able to discover it.

How can I configure the token and renovate bot to discover repositories without requiring project membership?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
I guess it's a result of the membership=true parameter here, which is hardcoded: https://github.com/renovatebot/renovate/blob/1f72816040aa5acc70a837eefc03d30b93ef6294/lib/modules/platform/gitlab/index.ts#L162-L170

This has been there since almost the beginning of Renovate GitLab support, added in https://github.com/renovatebot/renovate/pull/629, to solve https://github.com/renovatebot/renovate/issues/628

I wonder if the min_access_level=30 param is good enough that we don't need membership=true anymore. Even if so we should treat this as a breaking change in a major release just in case

**Reply by manuschillerdev** [reply]
Thanks for the super quick reply! 
I agree that this would be a breaking change. For a softer transition, we could make this configurable instead, and leave the default to true.

I am not familiar with the codebase, so I am probably missing places that need to be adjusted as well, but could sth like this conceptually work for you @rarkins ?

```diff
export interface AutodiscoverConfig {
+ membership?: boolean;
  topics?: string[];
  sort?: RepoSortMethod;
  order?: SortMethod;
  includeMirrors?: boolean;
  namespaces?: string[];
  projects?: string[];
}

const queryParams: Record<string, any> = { 
+  membership: config.membership ?? true, 
   per_page: 100, 
   with_merge_requests_enabled: true, 
   min_access_level: 30, 
   archived: false, 
 };
```

**Reply by rarkins** [reply]
I'd prefer to make it an experimental flag so that it doesn't require so much code changes and is more easy to remove

https://docs.renovatebot.com/self-hosted-experimental/
